### PR TITLE
fix(clickhouse): Set optimize setting to ensure ilike works

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

We upgraded CH recently, which exposed this bug: https://github.com/ClickHouse/ClickHouse/issues/36279

This PR is a workaround to ensure things keep working smoothly, before we can upgrade to a new version

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Unit tests

### CODE REVIEW

File: ee/clickhouse/queries/test/test_trends.py [hunk 1]
```diff
@@ -1561,3 +1561,59 @@ def test_same_day(self):
             self.team,
         )
         self.assertEqual(response[0]["data"], [1.0])
+
+    @test_with_materialized_columns(event_properties=["email", "name"], person_properties=["email", "name"])
+    def test_ilike_regression_with_current_clickhouse_version(self):
```
Comment (macobo) [inline ee/clickhouse/queries/test/test_trends.py:1566]: Q: is there a potential simpler repo in property.py tests?

Not urgent, but this is hard to follow

File: posthog/client.py [hunk 2]
```diff
@@ -152,8 +152,14 @@ def sync_execute(query, args=None, settings=None, with_column_types=False, flush
...
 
         try:
+            # Optimize_move_to_prewhere setting is set because of this regression test
+            # test_ilike_regression_with_current_clickhouse_version
+            # https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/test/test_trends.py#L1566
             result = client.execute(
-                prepared_sql, params=prepared_args, settings=settings, with_column_types=with_column_types
+                prepared_sql,
+                params=prepared_args,
+                settings={"optimize_move_to_prewhere": 0, **(settings or {}),},
```
[the remark this answers is not in the document]
Comment (macobo) [reply posthog/client.py]: I meant extract to top level of this file! :)

**Review by macobo — APPROVED** [review]
Small nit

File: posthog/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (neilkakkar) [reply posthog/client.py]: Is that because top of file = top of mind  / easy to find?

[the remark this answers is not in the document]
Comment (macobo) [reply posthog/client.py]: Yep - not n levels deep in some overtly complicated call stack.

**Comment by neilkakkar** [comment]
failing is a flake, kafka inserts. Merging!
